### ISSUE TITLE
Kanboard queue worker stops working after database error

### ISSUE THREAD

**Issue by rafacamargo123** [kanboard/kanboard#4169]
I have kanboard set up with beanstalkd and a unit (systemctl) created to handle the kanboard worker (kanboard/cli worker)
I noted that everyday in the morning the worker process stops working, the webhook is not called, the automatic actions neither.
It's the same problem of the issue #3716.
Apparently, the database server closes the connection after some inactive time.
```
2019-04-04T04:51:19.032258Z 2849977 [Note] Aborted connection 2849977 to db: 'kanboard' user: 'kanboard' host: 'touros.trt15.jus.br' (Got timeout reading communication packets)
```

It's easy to reproduce this issue on mysql:

1. Setup kanboard with beanstalkd and start the worker
1. Setup a project and create an automatic action that assign a color different from default when a task is created
1. Create a task to make sure the automatic action is working
1. On mysql server, identify the worker connection by running `select * from information_schema.processlist where db='kanboard';`, changing the db name if needed.
1. Kill the mysql connection by running `KILL id`, replacing id with the id returned on first query
1. Create a new task and see that the automatic action is no longer working

The instant you kill the connection, this erros appears on kanboard debug log:

```
[2019-04-04 13:36:31] [error] Kanboard\Core\Queue\JobHandler::executeJob: Error during job execution: SQL Error: SQLSTATE[HY000]: General error: 2006 MySQL server has gone away
[2019-04-04 13:36:31] [error] Kanboard\Core\Queue\JobHandler::executeJob => {"class":"Kanboard\\Job\\TaskEventJob","params":[8616,["task.create_update","task.create"],[],[],[]],"user_id":8}
```
I also noted the "time" field from the information_schema.processlist resets to 0 everytime an action is made on kanboard that makes the worker do something.
So, thats why I think this sudden lost of connectivity every day in the night is due to site inactivity.

I've setup the systemd unit to restart the worker service everytime it exits with failure, but the php process does not exit when there is a database exception.

I've solved this on my server by adding an exit(255);  at the end of the catch on executeJob() method of app/Core/Queue/JobHandler.php.
This way, the process will exit with failure status and the systemd will restart it.
The queue message will not be lost because it is removed only after this method completion, on app/Core/Queue/QueueManager.php listen() method: `$this->queue->completed($job);`

This simple solution worked for me, but I don't know if it's the best solution. Perhaps throw this exception up to the app/Console/WorkerCommand.php and exit there.

**Comment by fguillot** [comment]
Similar to #4152

**Comment by fguillot** [comment]
Duplicate of #4152
